File: resources/loop.rb
class Loop
  def number_of_older_residents(residents, residences)
    old_timers = (residents * residences).select do |res, resc|
      res.residence_id == resc.id && res.years_of_residence > resc.age
    end
    number_of_old_timers = old_timers.size
  end
end

File: resources/powerful/Loop.java
package powerful;

import java.util.List;


public class Loop {
    public int numberOfOlderResidents(List<Resident> residents, List<Residence> residences) {
        int numberOfOldTimers = 0;
        for (Resident res: residents) {
            for (Residence resc: residences) {
                if (res.getResidentId().equals(resc.getId()) && res.getYearsOfResidence().equals(resc.getAge())) {
                    ++numberOfOldTimers;
                }
            }
        }
        return numberOfOldTimers;
    }
}
